packages/entities/entities-shared/src/composables/useFetcher.ts

```diff
@@ -46,7 +49,15 @@ export default function useFetcher(
         requestUrl = `${requestUrl}&page[size]=${fetcherParams.pageSize}&page[number]=${fetcherParams.page}`
       }
 
-      const { data } = await axiosInstance.get(requestUrl)
+      const res = await axiosInstance.get(requestUrl)
+      // If the host app treats 404 responses as resolved through its validateStatus
+      // we need to throw the response to trigger the catch block
+      // because we still need the FetcherStatus.NoResults status
+      if (res.status === 404) {
```

Comment:
issue: this will only account for a `404`. 

What if the `validateStatus` argument wants to handle additional `4xx` status codes?

Reply:
Updated.